documentation/restarting.markdown

```diff
@@ -17,13 +17,29 @@ There are several ways to send a HUP signal to the Puppet Server process, but th
 
     kill -HUP `pgrep -f puppet-server`
 
+Starting in version 2.7.0, you can also reload Puppet Server by running the "reload" action via the operating system's service framework.  This is analogous to sending a hangup signal but with the benefit of having the "reload" command pause until the server has been completely reloaded, similar to how the "restart" command pauses until the service process has been fully restarted.
```

Comment:
> Why is this a benefit? I think I know why--it's safer in terms of stability and behaves more consistently--but I'm not sure that the benefit to users is clear here.

I see at least three clear advantages / cases where you'd want to use the `service reload` instead of `kill -HUP`:

1) In the context of automated scripting/tooling, `reload` more robustly helps you ensure when the server is back up and using any configuration changes that were just applied.

For example, you might have a script that does something like this:

- Change a setting in one of the puppetserver config files
- Reload the server
- Perform an agent run against the master

If the reload step were done via HUP, the server might not be up and running with the new configuration by the time the agent run is performed, possibly leading the agent run to fail or unintentionally perform actions based on the old configuration of the server.  If the reload were done via a `service reload` instead, though, the agent run should be able to contact the server and the server would assuredly be using the new configuration - leading to a more reliable, predictable result for the agent.

2) If the configuration change would cause the server to fail to reload properly - e.g., because the value for a setting were unintentionally fat-fingered - the `service reload` command would fail with an error exit code whereas the `kill -HUP` would give no indication of a failure.

3) It's arguably a bit easier to use the `service reload` than the `kill -HUP` approach because, with the `service reload` approach, the PID of the running puppetserver process does not need to be known in advance.

Granted, the `pgrep -f puppet-server` part would still allow you to dynamically determine the PID.  The whole `kill -HUP $(pgrep -f puppet-server)` command just seems a bit more onerous to me to remember than is `service puppetserver reload`.

Do you think any of the above would be useful to call out here?  I didn't want to be unnecessarily wordy here if some of these benefits would be obvious to most users.

> Also, if this is the preferred way of doing it, we should either bump the reload command process before the kill -HUP command, or just remove kill -HUP if we really don't want people to use it.

I wouldn't say that we don't want users to use the `kill -HUP` approach.  In most references that I've seen users make around the desire for service configuration changes to be reloaded without requiring a process restart, the request was for the ability to send a HUP signal to cause the reload, e.g., [SERVER-15](https://tickets.puppetlabs.com/browse/SERVER-15).  I suspect for many of our users, `kill -HUP` may still be more intuitive to use than `service puppetserver reload` - at least for one-off configuration reloads during development.  I don't think we should remove the `kill -HUP` documentation outright.  I'd be good with saying something about how we recommend the use of `service reload` in automated tooling because it more reliably determines when/whether the reload is successful and/or put the `service reload` section before the `HUP` section if you think that makes more sense, though.

@gguillotte - WDYT?  Happy to reword this however you think would be best for users.